msgcntr/messageforums-app/src/webapp/jsp/discussionForum/message/dfViewThreadDirect.jsp

```diff
@@ -42,21 +42,22 @@
 		return;
 
 	}else if(forumTool.getHasTopicAccessPrivileges(request.getParameter("topicId"))){
-		target = "/jsp/discussionForum/message/dfViewThread.jsf?messageId="
+		String forumsToolId = null;
+		try {
+			String siteId = ToolManager.getCurrentPlacement().getContext();
+			if (SiteService.getSite(siteId).getToolForCommonId("sakai.forums") != null) {
+				forumsToolId = SiteService.getSite(siteId).getToolForCommonId("sakai.forums").getId();
+			}
+		}
+		catch (Exception e) {
+			log.warn("dfViewThreadDirect - error while trying to get Forums tool id : {}", e.getMessage());
```

Comment:
just logging the message and then continuing to use a null forumsToolId is just going to cause more problems there should be better handling here.

Reply:
I've added a 'return' in the catch